File: src/main/java/burrows/apps/example/gif/presentation/main/IMainPresenter.java
package burrows.apps.example.gif.presentation.main;

import burrows.apps.example.gif.presentation.IBasePresenter;

/**
 * @author <a href="mailto:jaredsburrows@gmail.com">Jared Burrows</a>
 */
interface IMainPresenter extends IBasePresenter {
  void clearImages();
  void loadTrendingImages(Float next);
  void loadSearchImages(String searchString, Float next);
}

File: src/main/kotlin/burrows/apps/example/gif/presentation/main/IMainPresenter.kt
package burrows.apps.example.gif.presentation.main

import burrows.apps.example.gif.presentation.IBasePresenter

/**
 * @author [Jared Burrows](mailto:jaredsburrows@gmail.com)
 */
interface IMainPresenter : IBasePresenter {
  fun clearImages()
  fun loadTrendingImages(next: Int)
  fun loadSearchImages(searchString: String, next: Int)
}
